cla-backend-go/signatures/repository.go

```diff
@@ -2147,6 +2183,57 @@ func (repo repository) UpdateApprovalList(ctx context.Context, claManager *model
 			expressionAttributeValues[":go"] = attrList
 			updateExpression = updateExpression + " #GO = :go, "
 		}
+
+		if params.RemoveGithubOrgApprovalList != nil {
+			var invalidateErr error
+			approvalList.Criteria = utils.GitHubOrgCriteria
+			approvalList.ApprovalList = params.RemoveGithubOrgApprovalList
+			approvalList.Action = utils.RemoveApprovals
+			// Get repositories by CLAGroup
+			repositories, err := repo.repositoriesRepo.GetRepositoriesByCLAGroup(ctx, projectID, true)
+			if err != nil {
+				msg := fmt.Sprintf("unable to fetch repositories for claGroupID: %s ", projectID)
+				log.WithFields(f).Warn(msg)
+				return nil, errors.New(msg)
+			}
+			var ghOrgRepositories []*models.GithubRepository
+			var ghOrgs []*models.GithubOrganization
+			for _, repository := range repositories {
+				// Check for matching organization name in repositories table against approvalList removal GH Orgs
+				if utils.StringInSlice(repository.RepositoryOrganizationName, approvalList.ApprovalList) {
+					ghOrgRepositories = append(ghOrgRepositories, repository)
+				}
+			}
+
+			for _, ghOrgRepo := range ghOrgRepositories {
+				ghOrg, err := repo.ghOrgRepo.GetGithubOrganization(ctx, ghOrgRepo.RepositoryOrganizationName)
+				if err != nil {
+					msg := fmt.Sprintf("unable to get gh org by name: %s ", ghOrgRepo.RepositoryOrganizationName)
+					log.WithFields(f).Warn(msg)
+					return nil, errors.New(msg)
+				}
+				ghOrgs = append(ghOrgs, ghOrg)
+			}
+
+			var ghUsernames []string
+			for _, ghOrg := range ghOrgs {
+				ghOrgUsers, err := github.GetOrganizationMembers(ctx, ghOrg.OrganizationName, ghOrg.OrganizationInstallationID)
+				if err != nil {
+					msg := fmt.Sprintf("unable to fetch ghOrgUsers for org: %s ", ghOrg.OrganizationName)
+					log.WithFields(f).Warnf(msg)
+					return nil, errors.New(msg)
+				}
+				ghUsernames = append(ghUsernames, ghOrgUsers...)
```

Comment:
Is there a chance that we will have duplicates in this list? If so, let's prune for distinct values.